distribution/lib/Standard/Database/0.0.0-dev/src/DB_Table.enso

```diff
@@ -678,14 +678,16 @@ type DB_Table
                    new_filters = self.context.where_filters + [filter_column.expression]
                    new_ctx = self.context.set_where_filters new_filters
                    self.updated_context new_ctx
-           case filter of
+
+           filter_condition = case filter of
+               _ : Function ->  Panic.catch Any (filter:Filter_Condition) _->filter
+               _ -> filter
```

Comment:
As I was suggesting in call, I think it would be good to extract this into a separate helper function. Not only I expect this pattern will appear in more places, it will make it easier to understand why the pattern is needed to begin with.

```
## PRIVATE
   Special handling is needed for type `Filter_Condition | Function` to be able to support autoscoped constructors for `Filter_Condition`.
   The problem is, an autoscope constructor is considered a `Function` and thus it is _already_ deemed to be valid type, and the autoscope resolution fails to be triggered. Thus we need to try triggering it manually, with a fallback for other functions that were not the autoscoped constructor.
Filter_Condition.resolve_autoscoped (x : Filter_Condition | Function) = ...
```